### PR TITLE
[clinical-data-grid] Added callback ref for vertical and horizontal overflow container 

### LINKED ISSUE #580: [clinical-data-grid] Ability to launch clinical data grid with specific scroll positioning
# Feature Request

## Description

This request is to expose a ref callback prop on the DataGrid to expose the overflow container. Most solutions seems to be displaying things chronologically. If that is the case I think we will want to show most recent data face up. Current default behavior does not support this. It will be nice if clinical data grid provides this flexibility of default scroll right vs scroll left. Only option we have now is to use CSS dependencies on internal class names.

Two other related requests 
1. Ability to hide scroll bar. In future we see a need to control horizontal scrolling using external controls.
2. Ability to detect Item visibility. Use case- when there are selections outside the view area in the data grid, we would like to notify the user that there are items selected outside the view area. 

## @ Mentions

@neilpfeiffer
@tbiethman 
@chaitu9916

### PR DESCRIPTION
**Pull request by pranav300**
### Summary
Resolves #580

### CODE REVIEW

File: packages/terra-clinical-data-grid/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,8 @@ Changelog
 
 Unreleased
 ----------
+### Added
+* Added prop `overflowContainerRefCallback`.
```
Comment (nramamurth) [inline packages/terra-clinical-data-grid/CHANGELOG.md]: I'd be a bit more descriptive — as in, what does it do?

File: packages/terra-clinical-data-grid/tests/jest/DataGrid.test.jsx [hunk 2]
```diff
@@ -291,6 +292,23 @@ it('should render a DataGrid with the fill prop missing', () => {
   expect(dataGrid).toMatchSnapshot();
 });
 
+it('should pass in refCallback as the ref prop of the input element', () => {
```
Comment (nramamurth) [inline packages/terra-clinical-data-grid/tests/jest/DataGrid.test.jsx]: input element?

File: packages/terra-clinical-data-grid/src/DataGrid.jsx [hunk 3]
```diff
@@ -1084,7 +1089,12 @@ class DataGrid extends React.Component {
         >
           <div
             className={cx('vertical-overflow-container')}
-            ref={this.setVerticalOverflowContainerRef}
+            ref={(ref) => {
```
Comment (nramamurth) [inline packages/terra-clinical-data-grid/src/DataGrid.jsx]: @habelk, do these code change address your ask? 

> This request is to expose a ref callback prop on the DataGrid to expose the overflow container.

File: packages/terra-clinical-data-grid/tests/jest/DataGrid.test.jsx (hunk 2 above)
Comment (pranav300) [reply packages/terra-clinical-data-grid/tests/jest/DataGrid.test.jsx]: Fixed here https://github.com/cerner/terra-clinical/pull/585/commits/f40141f133c81d20aaa0b6004cb721b5c311ac71.

File: packages/terra-clinical-data-grid/CHANGELOG.md (hunk 1 above)
Comment (pranav300) [reply packages/terra-clinical-data-grid/CHANGELOG.md]: Fixed here https://github.com/cerner/terra-clinical/pull/585/commits/f40141f133c81d20aaa0b6004cb721b5c311ac71.

File: packages/terra-clinical-data-grid/src/DataGrid.jsx (hunk 3 above)
Comment (tbiethman) [reply packages/terra-clinical-data-grid/src/DataGrid.jsx]: We probably want to expose a ref to the 'horizontal-overflow-container' below as well. That will allow consumers to manipulate both the vertical and horizontal scrolling.

Comment (pranav300) [reply packages/terra-clinical-data-grid/src/DataGrid.jsx]: Updated here https://github.com/cerner/terra-clinical/pull/585/commits/b069773ed557b1c21ab850266334bffe3dfcc245.

Comment (habelk) [reply packages/terra-clinical-data-grid/src/DataGrid.jsx]: Yes, this satisfies our needs for scroll positioning. I will log a separate issue for other clinical data grid related enhancements based on https://github.com/cerner/terra-clinical/issues/580#issuecomment-545954846

File: packages/terra-clinical-data-grid/src/DataGrid.jsx [hunk 4]
```diff
@@ -1100,7 +1115,12 @@ class DataGrid extends React.Component {
             >
               <div
                 className={cx(['horizontal-overflow-container', { 'padded-container': fill }])}
-                ref={this.setHorizontalOverflowContainerRef}
+                ref={(ref) => {
+                  this.setHorizontalOverflowContainerRef(ref);
```
Comment (emilyrohrbough) [inline packages/terra-clinical-data-grid/src/DataGrid.jsx]: This logic should be added to [`this.setHorizontalOverflowContainerRef `](https://github.com/cerner/terra-clinical/blob/b069773ed557b1c21ab850266334bffe3dfcc245/packages/terra-clinical-data-grid/src/DataGrid.jsx#L549)
```diff
  setHorizontalOverflowContainerRef(ref) {
    this.setHorizontalOverflowContainerRef = ref;
+    if (this.props.horizontalOverflowContainerRefCallback) {
+       horizontalOverflowContainerRefCallback(ref)
+    }
  }

```

File: packages/terra-clinical-data-grid/src/DataGrid.jsx [hunk 5]
```diff
@@ -1084,7 +1094,12 @@ class DataGrid extends React.Component {
         >
           <div
             className={cx('vertical-overflow-container')}
-            ref={this.setVerticalOverflowContainerRef}
+            ref={(ref) => {
+              this.setVerticalOverflowContainerRef(ref);
+              if (verticalOverflowContainerRefCallback) {
```
Comment (emilyrohrbough) [inline packages/terra-clinical-data-grid/src/DataGrid.jsx]: This logic should be added to [`this.setVerticalOverflowContainerRef`](https://github.com/cerner/terra-clinical/blob/b069773ed557b1c21ab850266334bffe3dfcc245/packages/terra-clinical-data-grid/src/DataGrid.jsx#L573)
```diff
  setVerticalOverflowContainerRef(ref) {
    this.verticalOverflowContainerRef = ref;
+    if (this.props.verticalOverflowContainerRefCallback) {
+       verticalOverflowContainerRefCallback(ref)
+    }
  }

```

File: packages/terra-clinical-data-grid/src/DataGrid.jsx (hunk 4 above)
Comment (pranav300) [reply packages/terra-clinical-data-grid/src/DataGrid.jsx]: Updated here https://github.com/cerner/terra-clinical/pull/585/commits/9de4536895fbfa7f35601a8b54b810108df7b7da.

File: packages/terra-clinical-data-grid/src/DataGrid.jsx (hunk 5 above)
Comment (pranav300) [reply packages/terra-clinical-data-grid/src/DataGrid.jsx]: Updated here https://github.com/cerner/terra-clinical/pull/585/commits/9de4536895fbfa7f35601a8b54b810108df7b7da.

File: packages/terra-clinical-data-grid/src/DataGrid.jsx (hunk 4 above)
Comment (emilyrohrbough) [reply packages/terra-clinical-data-grid/src/DataGrid.jsx]: Can we revert to 
```js
 ref={this.setHorizontalOverflowContainerRef}
```
with this change?

Comment (pranav300) [reply packages/terra-clinical-data-grid/src/DataGrid.jsx]: updated here https://github.com/cerner/terra-clinical/pull/585/commits/aac516731bb55f93b6f08d36db934d9cb68a56dd.

**Pull request merged**